Added Forced platform

maine98 opened the pull request:
https://user-images.githubusercontent.com/46499668/103581972-45293700-4edd-11eb-8e0e-a499e0427492.mp4

[FIX]
   - Added forced platform settings

[ADD]
   - Added CustomControlsPage and custom controls component

jhomlala commented on example/.vscode/launch.json:
```diff
@@ -0,0 +1,13 @@
+{
```
Please remove this file.

jhomlala commented on example/lib/pages/custom_controller_page.dart:
```diff
@@ -0,0 +1,100 @@
+}
+
+class _CustomControllerPageState extends State<CustomControllerPage> {
+  BetterPlayerController _betterPlayerControllerMaterial;
```
Can you try do it with only one controller? User can change player style with 3 buttons.

jhomlala commented on example/lib/pages/custom_controller_page.dart:
```diff
@@ -0,0 +1,100 @@
+    );
+  }
+
+  void _initPlayers() {
```
Please move it above build method.

jhomlala commented on example/lib/pages/custom_controller_page.dart:
```diff
@@ -0,0 +1,100 @@
+
+  void _initPlayers() {
+    String url =
+        'https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4';
```
Please use urls from constants class.

jhomlala commented on example/lib/pages/welcome_page.dart:
```diff
@@ -77,6 +78,9 @@ class _WelcomePageState extends State<WelcomePage> {
       _buildExampleElementWidget("Controls configuration", () {
         _navigateToPage(ControlsConfigurationPage());
       }),
+      _buildExampleElementWidget("Custom Controls Player", () {
```
Move it to the bottom of the list. In the future I will re-order list of examples based on priority.

jhomlala commented on lib/src/configuration/better_player_configuration.dart:
```diff
@@ -1,14 +1,8 @@
-// Flutter imports:
```
Revert these changes please. I don't lnow what's the purpose of this change.

jhomlala commented on lib/src/controls/better_player_controls_configuration.dart:
```diff
@@ -76,7 +77,10 @@ class BetterPlayerControlsConfiguration {
 
   ///Custom controls, it will override Material/Cupertino controls
-  final Widget customControls;
+  final Widget Function(BetterPlayerController controller) customControls;
```
final Widget Function(BetterPlayerController) customControlsBuilder seems to be better name for me.

jhomlala commented on lib/src/controls/better_player_controls_configuration.dart:
```diff
@@ -76,7 +77,10 @@ class BetterPlayerControlsConfiguration {
+  final Widget Function(BetterPlayerController controller) customControls;
+
+  ///Flag to use a forced platform
+  final PlayerPlatform playerPlatform;
```
Basically I think that we shouldn't configure player theme based on platform rather than platform theme.

jhomlala replied:
Also please set default theme based on platform i.e.:
PlayerTheme.material for Android
PlayerTheme.cupertino for iOS

jhomlala commented on lib/src/controls/better_player_controls_configuration.dart:
```diff
@@ -193,3 +198,5 @@ class BetterPlayerControlsConfiguration {
   }
 }
+
+enum PlayerPlatform { IOS, ANDROID }
```
Please change PlayerPlatform to PlayerTheme with values: `Material,` `Cupertino`. Please move it to another class.

jhomlala commented on lib/src/core/better_player_with_controls.dart:
```diff
@@ -158,18 +158,40 @@ class _BetterPlayerWithControlsState extends State<BetterPlayerWithControls> {
+      if (controlsConfiguration.customControls == null &&
+          controlsConfiguration.playerPlatform == null) {
+        if (Platform.isAndroid) {
+          return BetterPlayerMaterialControls(
```
Please create 2 methods which will create Material and Cupertino controls when requested. I believe we have code duplication here.

jhomlala replied:
Also based on comment above, when we set PlayerTheme default value for given platform, we don't need to react here. So we dont  this if block:
``` if (controlsConfiguration.showControls) {
      if (controlsConfiguration.customControls == null &&
          controlsConfiguration.playerPlatform == null) {
        if (Platform.isAndroid) {
...
```

So we need only this if block:
```
if (showControls){
	if (customControls){
		return buildCustomControls();
	} else if (platformTheme == material){
		return buildMaterialControls();
	} else if (platformTheme == cupertino){
		return buildCupertinoControls();
	}
} 
```

jhomlala commented:
Thank you for PR. This seems to be useful, but you need to fix some code.

maine98 commented:
https://user-images.githubusercontent.com/46499668/103800993-63ae4000-504d-11eb-8540-1f57fb4db7e1.mp4




### [REMOVE]
   - Added .vscode/ from .gitignore
### [ADD]
   - Added enum PlayerTheme in another file and edded export inside betterplayer.dart
   - Added on enum PlayerTheme value custom for use a custom theme for controls player
### [REFACTOR]
   - Refactor page and file "custom_controls_page" in "change_player_theme_page"
   - Refactor file "CustomControlsWidget.dart" in "custom_controls_widget.dart"
### [FIX]
   - Fixed page "custom_controls_page" for using a only one controller for betterPlayer
   - Cleanup and reformat code
### [REVERT]
   - Revert modify inside "better_player_configurations.dart"

jhomlala commented:
LGTM, Thanks.
